src/libServer/WebsocketServer.cpp

```diff
@@ -0,0 +1,494 @@
+    return false;
+  } catch (...) {
+    LOG_GENERAL(WARNING, "other exception");
+    return false;
+  }
+
+  return true;
+}
+
+void WebsocketServer::stop() {
+  LOG_MARKER();
+
+  try {
+    // stopping the Websocket listener and closing outstanding connection
+    websocketpp::lib::error_code ec;
+    m_server.stop_listening(ec);
+    if (ec) {
+      LOG_GENERAL(WARNING,
+                  "websocket stop_listening failed, error: " << ec.message());
+      return;
+    }
+
+    {
+      // Close all existing websocket connections.
+      lock_guard<mutex> g(m_mutexTxBlockSockets);
+
+      for (auto& socket : m_txblock_websockets) {
+        websocketpp::lib::error_code ec;
+        m_server.close(socket.second, websocketpp::close::status::normal,
+                       "Terminating connection...", ec);
+        if (ec) {
+          LOG_GENERAL(WARNING, "websocket stop_listening (1) failed, error: "
+                                   << ec.message());
+        }
+      }
+    }
+
+    {
+      lock_guard<mutex> g(m_mutexEventLogSockets);
+
+      for (auto& socket : m_eventlog_websockets) {
+        websocketpp::lib::error_code ec;
+        m_server.close(socket.second, websocketpp::close::status::normal,
+                       "Terminating connection...", ec);
+        if (ec) {
+          LOG_GENERAL(WARNING, "websocket stop_listening (2) failed, error: "
+                                   << ec.message());
+        }
+      }
+    }
+
+    // Stop the end point
+    m_server.stop();
+    m_thread->join();
+  } catch (const websocketpp::exception& e) {
+    LOG_GENERAL(WARNING, "websocket stop failed, error: " << e.what());
+  } catch (...) {
+    LOG_GENERAL(WARNING, "other exception");
+  }
+  clean();
+}
+
+void WebsocketServer::clean() {
+  {
+    lock_guard<mutex> g(m_mutexTxBlockSockets);
+    m_txblock_websockets.clear();
+  }
+  {
+    lock_guard<mutex> g(m_mutexEventLogSockets);
+    m_eventlog_websockets.clear();
+    m_elsockettracker.clean();
+  }
+  {
+    lock_guard<mutex> g(m_mutexELDataBufferSockets);
+    m_eventLogDataBuffer.clear();
+  }
+  {
+    lock_guard<mutex> g(m_mutexEqIndex);
+    m_eqIndex.clear();
+  }
+}
+
+bool GetQueryEnum(const string& query, WEBSOCKETQUERY& q_enum) {
+  if (query == "NewBlock") {
+    q_enum = NEWBLOCK;
+  } else if (query == "EventLog") {
+    q_enum = EVENTLOG;
+  } else {
+    return false;
+  }
+
+  return true;
+}
+
+bool WebsocketServer::getWebsocket(const string& ip, WEBSOCKETQUERY query,
+                                   connection_hdl& hdl) {
+  IpSocketMap::iterator it;
+  switch (query) {
+    case NEWBLOCK:
+      it = m_txblock_websockets.find(ip);
+      if (it == m_txblock_websockets.end()) {
+        return false;
+      }
+      break;
+    case EVENTLOG:
+      it = m_eventlog_websockets.find(ip);
+      if (it == m_eventlog_websockets.end()) {
+        return false;
+      }
+      break;
+  }
+
```

Comment:
shall we add default case to handle invalid `it` by returning `false` ?

Reply:
since query only has two options it will never goes into the default case.